### PR TITLE
Adding eq for instructions

### PR DESCRIPTION
**Pull request by jaygambetta**
Needed to test if dag are equal is a way to test if instructions are equal.

### CODE REVIEW

**Comment by jaygambetta** [comment]
@ajavadia and @kdk done

**Comment by jaygambetta** [comment]
omg this is breaking the test which we intend to fix with the dag comparison.

File: test/python/circuit/test_instructions.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+# pylint: disable=unused-import
+
+"""Test Qiskit's QuantumCircuit class."""
```
Comment (ajavadia) [inline test/python/circuit/test_instructions.py]:
```suggestion
"""Test Qiskit's Instruction class."""
```

File: test/python/circuit/test_instructions.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import os
+import tempfile
+import unittest
+
+from qiskit import QuantumRegister, ClassicalRegister
+from qiskit._instruction import Instruction
+from qiskit.extensions.standard import h, cx
```
Comment (ajavadia) [inline test/python/circuit/test_instructions.py]:
```suggestion
from qiskit.extensions.standard.h import HGate
from qiskit.extensions.standard.cx import CnotGate
```

File: test/python/circuit/test_instructions.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        uop1 = Instruction('u', [0.4, 0.5, 0.5], qr, cr)
+        uop2 = Instruction('u', [0.4, 0.6, 0.5], qr, cr)
+        uop3 = Instruction('v', [0.4, 0.5, 0.5], qr, cr)
+        uop4 = Instruction('u', [0.4, 0.5, 0.5], qr, cr)
+        self.assertFalse(hop1 == hop2)
+        self.assertTrue(hop1 == hop3)
+        self.assertFalse(uop1 == uop2)
+        self.assertTrue(uop1 == uop4)
+        self.assertFalse(uop1 == uop3)
+        self.assertTrue(h == h)
```
Comment (ajavadia) [inline test/python/circuit/test_instructions.py]:
```suggestion
        self.assertTrue(HGate(qr[0]) == HGate(qr[1]))
```

File: test/python/circuit/test_instructions.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        self.assertFalse(h == cx)
```
Comment (ajavadia) [inline test/python/circuit/test_instructions.py]:
```suggestion
        self.assertFalse(HGate(qr[0]) == CnotGate(qr[0], qr[1]))
```

File: test/python/circuit/test_instructions.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        self.assertFalse(hop1 == h)
```
Comment (ajavadia) [inline test/python/circuit/test_instructions.py]:
```suggestion
        self.assertFalse(hop1 == HGate(qr[2]))
```

**Pull request merged**